src/definitions/ScannedTypehint.hack

```diff
@@ -14,6 +14,10 @@ use type Facebook\HHAST\{DotDotDotToken, InoutToken, Node, ResolvedTypeKind};
 
 /** Represents a parameter, property, constant, or return type hint */
 final class ScannedTypehint {
+
+  const UNION = '[union]';
+  const INTERSECTION = '[intersection]';
```

Comment:
This is kinda gross but it's the same as what we do for tuples -- we use "tuple" as a fake type name and then check for it below.